lib/spack/spack/ci.py

```diff
@@ -839,6 +1041,12 @@ def generate_gitlab_ci_yaml(
         else:
             broken_spec_urls = web_util.list_url(broken_specs_url)
 
+    spack_ci = SpackCI(ci_config, phases, staged_phases)
+    spack_ci_ir = spack_ci.generate_ir()
+
+    with open("spack_ci_ir.txt", "w") as f:
```

Comment:
Are you just writing out the ir for debugging purposes?  Should it be optional or always produced?